02-express-tutorial/02-http-app.js

```diff
@@ -0,0 +1,64 @@
+const http = require('http');
+const { readFileSync } = require('fs');
+
+//get all files
+const homePage = readFileSync('./navbar-app/index.html');
+const homeStyles = readFileSync('./navbar-app/styles.css');
+const homeImage = readFileSync('./navbar-app/logo.svg');
+const homeLogic = readFileSync('./navbar-app/browser-app.js');
+
+const server = http.createServer((req, res) => {
+   //if user visits our url we will serve him with data
+
+   // console.log(req.method)
+    const url = req.url;
+    console.log(url);
+
+  if (url === '/') {
+    //this is the home page
+    res.writeHead(200, { 'content-type': 'text/html' });
+    // if we set the header ourselves, without express, and instead of text/html we write text/plain , the result wil literally be <h1>home page</h1>
+    res.write(homePage); // here we are sending the body
+    //console.log('user hit the server');
+    res.end();
+    //This method signals to the server that all of the response headers
+    //and body have been sent; that server should consider this message complete. The method, response.end(),
+    // MUST be called on each response.
+  }
+  //about page
+  else if (url === '/about') {
+    res.writeHead(200, { 'content-type': 'text/html' });
+    res.write('<h1>About page</h1>'); // here we are sending the body
+    res.end();
+  }
+
+  //styles
+  else if (url === '/styles.css') {
+    res.writeHead(200, { 'content-type': 'text/css' });
+    res.write(homeStyles);
+    res.end();
+  }
+
+  //logo/image
+  else if (url === '/browser-app.js') {
+    res.writeHead(200, { 'content-type': 'image/svg+xml' });
+    res.write(homeImage);
+    res.end();
+  }
+
+  //logic
+  else if (url === '/logo.svg') {
+    res.writeHead(200, { 'content-type': 'text/javascript' });
+    res.write(homeLogic);
+    res.end();
+  }
```

Comment:
I think you have the urls for serving the logo vs the js logic mixed up here:

```suggestion
  //logo/image
  else if (url === '/logo.svg') {
    res.writeHead(200, { 'content-type': 'image/svg+xml' });
    res.write(homeImage);
    res.end();
  }

  //logic
  else if (url === '/browser-app.js') {
    res.writeHead(200, { 'content-type': 'text/javascript' });
    res.write(homeLogic);
    res.end();
  }
```

You'll notice that currently, if you try to load the home page, your code is leading to some errors:

![](https://user-images.githubusercontent.com/9094098/225613958-eab6db69-2275-4c41-9446-8b96b59982d7.png)

Reply:
Another thing you'll notice; is when the `browser-app.js` file isn't loaded correctly then the "hamburger" menu in the nav bar shown when the window is smaller won't work correctly:

![natalia_record_opt_2](https://user-images.githubusercontent.com/9094098/225625387-1c725bec-322c-4665-abdc-ce144182eed6.gif)

Reply:
my bad, thank you!!!